VmAgent.Core/Dependencies/BasicAssetExtractor.cs

```diff
@@ -0,0 +1,109 @@
+﻿using Microsoft.Azure.Gaming.VmAgent.Core.Interfaces;
+using System;
+using System.Collections.Generic;
+using System.Diagnostics;
+using System.IO;
+using System.Runtime.InteropServices;
+using System.Text;
+
+namespace Microsoft.Azure.Gaming.VmAgent.Core.Dependencies
+{
+    public class BasicAssetExtractor
+    {
+        public static BasicAssetExtractor Instance = new BasicAssetExtractor();
+
+        private readonly ISystemOperations _systemOperations;
+        private readonly MultiLogger _logger;
+        private string _zipExtension;
+        private string _tarExtension;
+
+        public BasicAssetExtractor(ISystemOperations systemOperations = null, MultiLogger logger = null, string zipExtension = null, string tarExtension = null)
+        { 
+            _systemOperations = systemOperations;
+            _logger = logger;
+            _zipExtension = zipExtension;
+            _tarExtension = tarExtension;
+        }
+
+        public void ExtractAssets(string assetFileName, string targetFolder)
+        {
+            // If the OS is windows use the native .NET zip extraction (we only support .zip for Windows).
+            if (_systemOperations.IsOSPlatform(OSPlatform.Windows))
```

Comment:
I'm not sure if we need to add the code here that checks if file asset name is ending with supported format in VmAgent (.tar, .tar.gz, .zip), because we already have validation code on front side (Control Plane, and Web Portal)

Reply:
The agent code was written before/separately from the front end. So we could theoretically remove the checks, but no harm in keeping it either.